### PR TITLE
fix(tools): update github actions and workflows

### PR DESCRIPTION
**Pull request by raisedadead**
This PR aims to:

- [x] Remove the merge-conflict action.
      The merge conflict action was giving more false positives than intented, we might want to try and make it a one-off workflow or approach it differently all together. Discussed and removed this with confirmation from Randell.

- [x] Adds a Unit Tests action (to transition away from Travis).
      The Travis CI has has heightened error rates with failing to report back status on PR. Often the run is green but the status remains waiting. I have added a new action in here to run the CI tests right here on GitHub instead. The Travis integration will be removed in a few days after reliability checks.

- [x] Have some uniformity on the Lighthouse CI and the Cypress Tests actions.
      The Cypress and Lighthouse checks needed production like stack and were doing setup and tear-down in different ways, this makes it a bit uniform.

- [x] Made additional misc. consistency updates on the workflows to name every step (better for logs) amongst other small tweaks.

Note: This PR was intentionally not made from my fork to avoid the pitfalls of GitHub and be closer to the real setup.

### CODE REVIEW

**Comment by raisedadead** [comment]
Tested that a job can prevent further runs, bringing parity with Travis.

image

**Comment by raisedadead** [comment]
I wanted to add on a wait on checks action in there but it seemed a bit unreliable. This should be good to once the run is green.

I will remove the Travis integration after a few days, if the actions work well.

File: .github/workflows/node.js-tests.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+          node-version: ${{ matrix.node-version }}
+
+      - name: Set Environment variables
+        run: cp sample.env .env
+
+      - name: Lint Source Files
+        run: |
+          npm ci
+          npm run ensure-env --if-present
+          npm run lint --if-present
```
Comment (ojeytonwilliams) [inline .github/workflows/node.js-tests.yml:33]: Don't we want them to fail if those scripts are missing?

Comment (raisedadead) [reply .github/workflows/node.js-tests.yml:33]: Probably yes, but I think that's a larger problem.

**Pull request merged**